File: src/main/java/random/telegramhomebot/auth/services/DefaultUserService.java
package random.telegramhomebot.auth.services;


import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import random.telegramhomebot.auth.entities.User;
import random.telegramhomebot.auth.repositories.UserRepository;

import javax.transaction.Transactional;
import java.util.Optional;

@RequiredArgsConstructor
@Service
@Transactional
public class DefaultUserService implements UserService {

    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;

    @Override
    public Optional<User> getUserByID(final long id) {
        return userRepository.findById(id);
    }

    @Override
    public void changeUserPassword(final User user, final String password) {
        user.setPassword(passwordEncoder.encode(password));
        userRepository.save(user);
    }

    @Override
    public boolean checkIfValidOldPassword(final User user, final String oldPassword) {
        return passwordEncoder.matches(oldPassword, user.getPassword());
    }
}

File: src/main/java/random/telegramhomebot/auth/services/DefaultUserService.kt
package random.telegramhomebot.auth.services

import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import random.telegramhomebot.auth.db.entities.User
import random.telegramhomebot.auth.db.repositories.UserRepository
import reactor.core.publisher.Mono
import javax.transaction.Transactional

@Service
@Transactional
class DefaultUserService(
    private val userRepository: UserRepository,
    private val passwordEncoder: PasswordEncoder
) : UserService {

    override fun getUserByID(id: Long): Mono<User> = userRepository.findById(id)

    override fun changeUserPassword(user: User, password: String) {
        user.password = passwordEncoder.encode(password)
        userRepository.save(user).block()
    }

    override fun checkIfValidOldPassword(user: User, oldPassword: String) =
        passwordEncoder.matches(oldPassword, user.password)
}
